src/protocolhandler.ts

```diff
@@ -225,40 +219,49 @@ export class ProtocolHandler extends EventEmitter {
         });
     }
 
-    publish(subject: string, data: any, reply: string = ""): void {
+    publish(subject: string, data: any, reply: string = ''): void {
         if (this.closed) {
             throw (NatsError.errorForCode(ErrorCode.CONN_CLOSED));
         }
+
         if (this.noMorePublishing) {
             throw NatsError.errorForCode(ErrorCode.CONN_DRAINING);
         }
+
         data = this.toBuffer(data);
-        let len = data.length;
+
+        const len = data.length;
         let proto: string;
         if (reply) {
             proto = `PUB ${subject} ${reply} ${len}`;
         } else {
             proto = `PUB ${subject} ${len}`;
         }
+
         this.sendCommand(this.buildProtocolMessage(proto, data))
     }
 
-    subscribe(s: Sub): Subscription {
+    subscribe(subscription: Sub): Subscription {
```

Comment:
this is not a subscription but a descriptor- the actual subscription is created at the bottom.